WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -167,15 +216,44 @@ public void onSitePluginUpdated(OnSitePluginUpdated event) {
                     + event.error.message);
             return;
         }
-        if (event.plugin != null) {
-            mPlugin = event.plugin;
-            refreshStates();
+        mPlugin = mPluginStore.getSitePluginByName(mSite, mPlugin.getName());
+        refreshViews();
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPluginInfoChanged(PluginStore.OnPluginInfoChanged event) {
+        if (event.isError()) {
+            AppLog.e(AppLog.T.API, "An error occurred while fetching the plugin info with type: "
+                    + event.error.type);
+            return;
+        }
+        if (event.pluginInfo != null && mPlugin.getSlug().equals(event.pluginInfo.getSlug())) {
+            mPluginInfo = event.pluginInfo;
+            refreshPluginVersionViews();
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSitePluginVersionUpdated(PluginStore.OnSitePluginVersionUpdated event) {
+        if (event.isError()) {
+            AppLog.e(AppLog.T.API, "An error occurred while updating the plugin version with type: "
+                    + event.error.type);
+            return;
         }
+        mPlugin = mPluginStore.getSitePluginByName(mSite, mPlugin.getName());
```

Comment:
Is it guaranteed that `mPluginStore.getSitePluginByName` will never return null? If not then we should make sure `mPlugin`  is never set to null here.

Reply:
I'd say it is, but I don't think it hurts us to check for it for edge cases. It's added in 8c5b1b28f17023e5bb5cf2d0dabf06a692eecb57